### PR TITLE
Added yaml file for nvme module to test driver_module_parameters

### PR DESCRIPTION
**Pull request by Naresh-ibm**
Added yaml file with different parameters and values for nvme module
to test different parameter of module by changing their values with
unload and load of module.

Signed-off-by: Naresh Bannoth <nbannoth@in.ibm.com>

### CODE REVIEW

**Comment by Naresh-ibm** [comment]
Run logs
---------------
[root@ltcever60-lp14 driver]# avocado run --test-runner runner driver_parameter_block_device.py -m driver_parameter_block_device.py.data/driver_parameter_block_device_nvme.yaml 
JOB ID     : 9991b81823ea068b625531851a3066ebddbefd03
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2022-03-27T06.42-9991b81/job.log
 (01/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-io_queue_depth-value-io_queue_depth_100-6f75: PASS (64.82 s)
 (02/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-io_queue_depth-value-io_queue_depth_2048-df7a: PASS (64.81 s)
 (03/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-io_queue_depth-value-io_queue_depth_5120-87b3: PASS (64.74 s)
 (04/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-io_queue_depth-value-io_queue_depth_10240-a949: PASS (64.77 s)
 (05/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-max_host_mem_size_mb-value-max_host_mem_size_mb_10-3feb: PASS (64.87 s)
 (06/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-max_host_mem_size_mb-value-max_host_mem_size_mb_64-6584: PASS (64.79 s)
 (07/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-max_host_mem_size_mb-value-max_host_mem_size_mb_256-1641: PASS (64.77 s)
 (08/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-max_host_mem_size_mb-value-max_host_mem_size_mb_1280-b06b: PASS (64.84 s)
 (09/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-noacpi-value-noacpi_Y-1834: PASS (64.91 s)
 (10/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-noacpi-value-noacpi_N-8204: PASS (64.74 s)
 (11/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-poll_queues-value-poll_queues_0-b9bb: PASS (64.86 s)
 (12/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-poll_queues-value-poll_queues_2-da9e: PASS (64.81 s)
 (13/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-poll_queues-value-poll_queues_4-ba5d: PASS (64.83 s)
 (14/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-poll_queues-value-poll_queues_8-8f2c: PASS (64.74 s)
 (15/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-sgl_threshold-value-sgl_threshold_1000-1852: PASS (64.77 s)
 (16/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-sgl_threshold-value-sgl_threshold_10000-7123: PASS (64.76 s)
 (17/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-sgl_threshold-value-sgl_threshold_32000-ca50: PASS (64.86 s)
 (18/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-sgl_threshold-value-sgl_threshold_64000-3374: PASS (64.83 s)
 (19/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-use_cmb_sqes-value-use_cmb_sqes_Y-99d4: PASS (64.86 s)
 (20/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-use_cmb_sqes-value-use_cmb_sqes_N-02d6: PASS (64.82 s)
 (21/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-write_queues-value-write_queues_0-1595: PASS (64.75 s)
 (22/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-write_queues-value-write_queues_2-3973: PASS (64.81 s)
 (23/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-write_queues-value-write_queues_4-f41a: PASS (64.85 s)
 (24/24) driver_parameter_block_device.py:Moduleparameter.test;run-Test-write_queues-value-write_queues_8-6c40: PASS (64.87 s)
RESULTS    : PASS 24 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2022-03-27T06.42-9991b81/results.html
JOB TIME   : 1556.34 s
[root@ltcever60-lp14 driver]#

**Comment by Naresh-ibm** [comment]
[driver_parameter_block_device_nvme_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/8357507/driver_parameter_block_device_nvme_job.log)

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
Remove 

multipath_enabled: False
disk: '/dev/nvme0n1'

File: io/driver/driver_parameter_block_device.py.data/driver_parameter_block_device_nvme.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+module: 'nvme'
+multipath_enabled: False
+disk: '/dev/nvme0n1'
```
[the remark this answers is not in the document]
Comment (Naresh-ibm) [reply io/driver/driver_parameter_block_device.py.data/driver_parameter_block_device_nvme.yaml:3]: Yes @abdhaleegit they are needed.
the disk parameter is used under dd method. please check that.
after each driver parameter we are running dd command to check the IO runs happens or not.
hence this is needed.

[the remark this answers is not in the document]
Comment (abdhaleegit) [reply io/driver/driver_parameter_block_device.py.data/driver_parameter_block_device_nvme.yaml:2]: ah ok I was looking at generic driver param code

**Review by abdhaleegit — APPROVED** [review]
LGTM

File: io/driver/driver_parameter_block_device.py.data/driver_parameter_block_device_nvme.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (abdhaleegit) [reply io/driver/driver_parameter_block_device.py.data/driver_parameter_block_device_nvme.yaml:2]: got it

**Pull request merged**